### PR TITLE
Add jsonschema exporter for VSS.

### PR DESCRIPTION
**Pull request by adobekan**
Work still in progress.

to try,
`./vspec2jsonschema.py ../spec/VehicleSignalSpecification.vspec ../../mydemo.schema.json`

### CODE REVIEW

File: vspec/vssexporters/vss2jsonschema.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    "int64[]": "array",
+    "uint64[]": "array",
+    "boolean[]": "array",
+    "float[]": "array",
+    "double[]": "array",
+    "string[]": "array"
+}
+
+def add_arguments(parser: argparse.ArgumentParser):
+    parser.description = "The JSON schema exporter does not support any additional arguments."
```
Comment (erikbosch) [inline vspec/vssexporters/vss2jsonschema.py:46]: Should we possibly have `--json-pretty` similar to vspec2json?

Comment (adobekan) [reply vspec/vssexporters/vss2jsonschema.py:46]: i guess it should work same as for JSON.

**Comment by adobekan** [comment]
Here is an OpenAPI example as well
```
openapi: 3.1.0
info:
  title: Vehicle Acceleration API
  version: 1.0.1
servers:
  - url: https://api.accelerationservice.com/v1

paths:
  /vehicle/acceleration:
    post:
      summary: Provide Vehicle Acceleration Data
      description: Endpoint to send vehicle acceleration data.
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: schema/mydemo_vehicle.schema.json#/Vehicle/$children/Acceleration
      responses:
        '200':
          description: Success. Acceleration data received.
        '400':
          description: Bad Request. Invalid payload or data format.

  /vehicle/acceleration/mySpecificLocationEvent:
    get:
      summary: Get Current Vehicle Data
      description: Get the current location and acceleration data of the vehicle.
      responses:
        '200':
          description: Success. Current vehicle data received.
          content:
            application/json:
              schema:
                type: object
                properties:
                  Location:
                    $ref: schema/mydemo_vehicle.schema.yaml#/Vehicle/$children/CurrentLocation
                  Acceleration:
                    $ref: schema/mydemo_vehicle.schema.yaml#/Vehicle/$children/Acceleration


  /vehicle/acceleration/TriggerService:
    get:
      summary: Get Current Vehicle Data
      description: Get the current location and acceleration data of the vehicle.
      parameters:
        - name: longitudinal_threshold
          in: query
          description: Threshold for longitudinal acceleration (X-axis) in m/s^2.
          schema:
            type: number
        - name: lateral_threshold
          in: query
          description: Threshold for lateral acceleration (Y-axis) in m/s^2.
          schema:
            type: number
      responses:
        '200':
          description: Success. Current vehicle data received.
          content:
             application/json:
              schema:
                type: object
                properties:
                  Timestamp:
                    type: string
                    format: date-time
                    description: The timestamp of the vehicle data.
                  Location:
                    $ref: schema/mydemo_vehicle.schema.yaml#/Vehicle/$children/CurrentLocation
                  Acceleration:
                    $ref: schema/mydemo_vehicle.schema.yaml#/Vehicle/$children/Acceleration
```

**Comment by SebastianSchildt** [comment]
See also https://wiki.covesa.global/display/WIK4/Data+Expert+Group+Workshops+-+Interface+Pillar+Alignment?preview=/71073952/75727000/COVESA_VSS_OpenAPI.pdf

**Comment by SebastianSchildt** [comment]
https://wiki.covesa.global/display/WIK4/Data+Expert+Group+Workshops+-+Interface+Pillar+Alignment

**Comment by erikbosch** [comment]
Meeting notes:

- Sebastian: Think this should be in 4.1, can volunteer if Adnan has no time
- Adnan: lets do it together

**Comment by erikbosch** [comment]
Commit sha: [0dc6a7f](https://github.com/COVESA/vss-tools/pull/296/commits/0dc6a7f06186dd26d760d95b1c1339dbdeab1300), Author: Adnan Bekan, Committer: Adnan Bekan; The sign-off is missing.

**Comment by erikbosch** [comment]
Two things I would like to have before merging it:

- Add the new tool to https://github.com/COVESA/vss-tools/blob/master/setup.py so that it gets included in PyPI packages
- Extend at least one of the testcases (like https://github.com/COVESA/vss-tools/tree/master/tests/vspec/test_datatypes) to include the tool, so that we make sure that it remains functional.

Another topic - do we want it to be included as a future release artifact? If so we should better update https://github.com/COVESA/vehicle_signal_specification/wiki/Release-Instructions-and-Checklist#create-output-artifacts and possibly the VSS Makefile as well to include it in builds/releases, but that can be done after this PR is merged

**Comment by SebastianSchildt** [comment]
I agree on the tests and setup.py, currently however I do think this tool does not work correctly yet. It seems to create something that is almost but not entirely unlike a json schema

Investigating

**Comment by SebastianSchildt** [comment]
The code dod not work with models that do not have `Vehicle` in the root (such as test cases and probably EV charging group. Fixed it)

**Comment by SebastianSchildt** [comment]
Added basic tests

**Comment by adobekan** [comment]
> The code dod not work with models that do not have `Vehicle` in the root (such as test cases and probably EV charging group. Fixed it)

Here i was 

> The code dod not work with models that do not have `Vehicle` in the root (such as test cases and probably EV charging group. Fixed it)

Was thinking about this one even in the case if you are defining API for only specific domain. Great.

**Comment by SebastianSchildt** [comment]
Added documentation.  

`--extended-attributes , -e` seems to be broken, seems to only works with  ` --jsonschema-all-extended-attributes`

Will investigate

**Comment by adobekan** [comment]
> Added documentation.
> 
> `--extended-attributes , -e` seems to be broken, seems to only works with ` --jsonschema-all-extended-attributes`
> 
> Will investigate

Did not try that option. Could it be space where it was defined, maybe we have some overlapping with other tools?

**Comment by SebastianSchildt** [comment]
Well. It works..... I think I mixed up some files when checking before. Marking this ready to review (or merge imo)

**Comment by erikbosch** [comment]
Meeting notes:

- Merge

**Pull request merged**